problem_builder/plot.py

```diff
@@ -0,0 +1,258 @@
+        if not self.claims:
+            return []
+
+        mentoring_block = self.get_parent().get_parent()
+        question_ids, questions = mentoring_block.question_ids, mentoring_block.questions
+        claims = []
+        for line in self.claims.split('\n'):
+            claim, q1, q2 = line.split(', ')
+            r1, r2 = None, None
+            for question_id, question in zip(question_ids, questions):
+                if question.name == q1:
+                    r1 = response_function(question, question_id)
+                if question.name == q2:
+                    r2 = response_function(question, question_id)
+                if r1 is not None and r2 is not None:
+                    break
+            claims.append([claim, r1, r2])
+
+        return claims
+
+    def _get_default_response(self, question, question_id):
+        from .tasks import _get_answer  # Import here to avoid circular dependency
+        # 1. Obtain block_type for question
+        question_type = question.scope_ids.block_type
+        # 2. Obtain latest submission for question
+        student_dict = {
+            'student_id': self.runtime.anonymous_student_id,
+            'course_id': self.course_key_str,
+            'item_id': question_id,
+            'item_type': question_type,
+        }
+        submissions = sub_api.get_submissions(student_dict, limit=1)
+        # 3. Extract response from latest submission for question
+        answer_cache = {}
+        for submission in submissions:
+            answer = _get_answer(question, submission, answer_cache)
+            return int(answer)
+
+    def _get_average_response(self, question, question_id):
+        from .tasks import _get_answer  # Import here to avoid circular dependency
+        # 1. Obtain block_type for question
+        question_type = question.scope_ids.block_type
+        # 2. Obtain latest submissions for question
+        submissions = sub_api.get_all_submissions(self.course_key_str, question_id, question_type)
+        # 3. Extract responses from latest submissions for question and sum them up
+        answer_cache = {}
+        response_total = 0
+        num_submissions = 0  # Can't use len(submissions) because submissions is a generator
+        for submission in submissions:
+            answer = _get_answer(question, submission, answer_cache)
+            response_total += int(answer)
+            num_submissions += 1
+        # 4. Calculate average response for question
+        if num_submissions:
+            return response_total / float(num_submissions)
+
+    def default_claims_json(self):
+        return json.dumps(self.default_claims)
+
+    def average_claims_json(self):
+        return json.dumps(self.average_claims)
+
+    @XBlock.json_handler
+    def get_data(self, data, suffix):
+        return {
+            'default_claims': self.default_claims,
+            'average_claims': self.average_claims,
+        }
+
+    def author_preview_view(self, context):
```

Comment:
It would be really nice if some sort of preview could be added, even with phony data somehow. Or just showing the empty grid with buttons. The answer data should be available on the studio side, I'd think-- so someone in the studio should be able to test with it directly if they really wanted to.